### DISCUSSION TITLE
question on how to set default relative directory when joining two files

### DISCUSSION THREAD

**Discussion by 7ticks** [mechatroner/vscode_rainbow_csv#99] [Q&A] [answered]
I can see that I need to provide either the full absolute path to the second file, which I use in the join clause, or I can provide relative path with ~ representine my home folder. I included an example of this path below. My quesiton is, "if all files that I use in join clauses are located in the same folder, is there an option to refer to these files simply by their names?' In other words, is it possible to set currently opened folder as "root folder" for this project, so that it would allow me to simply reference files names, instead of their absolute paths within my PC directory.

`select a1, a2, b2 inner join ~\OneDrive\Documents\backbone_ospf\two.csv on a1 == b1`

**Comment by mechatroner** [comment]
The only thing that would work right now is to open the other file in the VSCode and "name" it with `CTRL+P` ->  `Set join table name for RBQL`  command. If you name the other file `b` you would be able to use: `select a1, a2, b2 inner join b on a1 == b1`. And the column variables would still b1, b2, ... etc no matter how the join table is named.

Otherwise, I agree that something like this would be very convenient, I will try to think about a way to implement this feature. One problem is that typically relative paths are relative to the current working directory of the script.

**Comment by mechatroner** [comment]
@7ticks I thought a little more about your idea and decided that it is relatively safe to implement it in this way:
1. Try to find the JOIN file using the provided path as absolute or relative to the CWD.
2. If step 1 has failed and the path is relative try to find the JOIN file relative to the main input table directory.

I already submitted and deployed this change for the main RBQL project, Vim, and Sublime Text 3 extensions. And I am planning to integrate this new logic into the VSCode extension soonish. 

Thanks again for your proposal!

**Comment by mechatroner** [comment, marked as answer]
@7ticks I just published a new version 2.0.0 which allows to specify join path as relative to the main file directory. So this should work now.

**Comment by 7ticks** [comment]
Thank you very much Dmitry, I tested joining files from the same directory,
and it works as expected. I can simply specify their names without
specifying an absolute path for them. It makes the select statement a
little shorter and easier to write.

**Reply by mechatroner** [reply]
You are welcome! And thanks for suggesting this feature!
